business/istio_validations_test.go

```diff
@@ -712,52 +712,55 @@ func TestGetVSReferencesNotExisting(t *testing.T) {
 	assert.Nil(references)
 }
 
-func TestValidatingSingleObjectUpdatesList(t *testing.T) {
-	assert := assert.New(t)
-	require := require.New(t)
-	conf := config.NewConfig()
-	config.Set(conf)
-
-	vs := mockCombinedValidationService(t, conf, fakeIstioConfigList(),
-		[]string{"details.test.svc.cluster.local", "product.test.svc.cluster.local", "product2.test.svc.cluster.local", "customer.test.svc.cluster.local"})
-
-	v, err := vs.userClients[conf.KubernetesConfig.ClusterName].Istio().NetworkingV1().VirtualServices("test").Get(context.Background(), "product-vs", v1.GetOptions{})
-	require.NoError(err)
-
-	vInfo, err := vs.NewValidationInfo(context.Background(), []string{conf.KubernetesConfig.ClusterName}, nil)
-	require.NoError(err)
-	validationPerformed, validations, err := vs.Validate(context.Background(), conf.KubernetesConfig.ClusterName, vInfo)
-	require.NoError(err)
-	assert.True(validationPerformed)
-	vs.kialiCache.Validations().Replace(validations)
-
-	currentValidations, err := vs.GetValidations(context.Background(), conf.KubernetesConfig.ClusterName)
-	require.NoError(err)
-
-	key := models.IstioValidationKey{ObjectGVK: kubernetes.VirtualServices, Namespace: "test", Name: "product-vs"}
-	require.True(currentValidations[key].Valid)
-
-	v.Spec.Gateways = []string{"nonexistant"}
-	_, err = vs.userClients[conf.KubernetesConfig.ClusterName].Istio().NetworkingV1().VirtualServices("test").Update(context.Background(), v, v1.UpdateOptions{})
-	require.NoError(err)
-
-	// make sure validations are updated in a cache before retrieving them
-	vInfo, err = vs.NewValidationInfo(context.Background(), []string{conf.KubernetesConfig.ClusterName}, nil)
-	require.NoError(err)
-	validationPerformed, validations, err = vs.Validate(context.Background(), conf.KubernetesConfig.ClusterName, vInfo)
-	require.NoError(err)
-	assert.True(validationPerformed)
-	vs.kialiCache.Validations().Replace(validations)
-
-	updatedValidations, _, err := vs.ValidateIstioObject(context.Background(), conf.KubernetesConfig.ClusterName, "test", kubernetes.VirtualServices, "product-vs")
-	require.NoError(err)
-	require.False(updatedValidations[key].Valid)
-
-	validations, err = vs.GetValidations(context.Background(), conf.KubernetesConfig.ClusterName)
-	require.NoError(err)
-	require.NotEmpty(validations)
-	assert.False(validations[key].Valid)
-}
+// TODO: This test is currently broken because the testing cache uses a different
+// fake client and object trackers are different. Add this test back in once both
+// clients are the same.
```

Comment:
@nrfox I realize this is already merged, I just wanted to make sure this, and a few tests like it, were still supposed to be disabled?

Reply:
We either need to rewrite the test or use controller-runtime fake client instead of the kube fake clients. I can go back and see how feasible it is to use the same fake client and if it's too much work I'll just rewrite the test.